docs/reference/elasticsearch/mapping-reference/dense-vector.md

```diff
@@ -281,6 +281,9 @@ $$$dense-vector-index-options$$$
     :   In case a knn query specifies a `rescore_vector` parameter, the query `rescore_vector` parameter will be used instead.
     :   See [oversampling and rescoring quantized vectors](docs-content://solutions/search/vector/knn.md#dense-vector-knn-search-rescoring) for details.
 :::::
+
+`early_termination`
+:   (Optional, boolean) Apply _patience_ based early termination strategy to knn queries over HNSW graphs (see [paper](https://cs.uwaterloo.ca/~jimmylin/publications/Teofili_Lin_ECIR2025.pdf)). This is expected to produce Only applicable to `hnsw`, `int8_hnsw`, `int4_hnsw` and `bbq_hnsw` index types.
```

Comment:
I think this should be an index wide setting if possible. It can be dynamic there since its query time only, but having it per vector field seems like too much.

Reply:
I can see your point, however I can think of this setting to be either enabled or not either index or field-wise, whereas I don't think users would enable/disable it on a per query basis.

Reply:
>  however I can think of this setting to be either enabled or not either index or field-wise

I agree, I think it should be done at the index level. I only mention the "query time" nature of it to indicate that it can be a dynamic index setting.